static/src/javascripts/projects/common/modules/commercial/track-ad.js

```diff
@@ -0,0 +1,30 @@
+define([
+    'Promise',
+    'common/utils/mediator'
+], function(Promise, mediator) {
+    return trackAd;
+
+    function trackAd(id) {
+        return new Promise(function (resolve, reject) {
+            var onAdLoaded = function (event) {
+                if (event.slot.getSlotElementId() === id) {
+                    unlisten();
+                    resolve(!event.isEmpty);
```

Comment:
Do we have a use case for that? I'm wary of storing in memory objects we don't really control, such as the [`Slot`](https://developers.google.com/doubleclick-gpt/reference#googletagslot) object in this case.